File: server/src/entities/productImage/ProductImage.ts
import {
    Entity,
    PrimaryGeneratedColumn,
    Column,
    ManyToOne,
    JoinColumn,
} from 'typeorm';
import Product from '../product/Product';

@Entity({ name: 'product_images' })
export default class ProductImage {
    @PrimaryGeneratedColumn()
    id!: number;

    @Column('varchar', { length: 255 })
    src!: string;

    @Column('decimal')
    width!: number;

    @Column('decimal')
    height!: number;

    @ManyToOne(() => Product, (type) => type.images)
    @JoinColumn({ name: 'product_id' })
    product!: Product;
}

File: scraper/entities/productImage/ProductImage.py
from lib.database.createDatabaseConnection import db
from pony.orm import PrimaryKey, Required, Optional


class ProductImage(db.Entity):
    _table_ = 'product_images'
    id = PrimaryKey(int, auto=True)
    src = Required(str)
    width = Required(int)
    height = Required(int)
    product_id = Optional('Product')
